collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -1821,91 +1823,97 @@ private boolean saveForm(boolean exit, boolean complete, String updatedSaveName,
-                    showLongToast(String.format(getString(R.string.encryption_error_message),
-                            result.getMessage()));
-                    finishAndReturnInstance();
-                    break;
+                showLongToast(message);
+                break;
 
-                case CONSTRAINT_ERROR: {
-                    DialogUtils.dismissDialog(SaveFormProgressDialogFragment.class, getSupportFragmentManager());
-                    refreshCurrentView();
+            case FINALIZE_ERROR:
+                DialogUtils.dismissDialog(SaveFormProgressDialogFragment.class, getSupportFragmentManager());
+                showLongToast(String.format(getString(R.string.encryption_error_message),
+                        result.getMessage()));
+                finishAndReturnInstance();
+                break;
 
-                    // get constraint behavior preference value with appropriate default
-                    String constraintBehavior = (String) GeneralSharedPreferences.getInstance()
-                            .get(GeneralKeys.KEY_CONSTRAINT_BEHAVIOR);
+            case CONSTRAINT_ERROR: {
+                DialogUtils.dismissDialog(SaveFormProgressDialogFragment.class, getSupportFragmentManager());
+                refreshCurrentView();
 
-                    // an answer constraint was violated, so we need to display the proper toast(s)
-                    // if constraint behavior is on_swipe, this will happen if we do a 'swipe' to the
-                    // next question
-                    if (constraintBehavior.equals(GeneralKeys.CONSTRAINT_BEHAVIOR_ON_SWIPE)) {
-                        next();
-                    } else {
-                        // otherwise, we can get the proper toast(s) by saving with constraint check
-                        saveAnswersForCurrentScreen(EVALUATE_CONSTRAINTS);
-                    }
+                // get constraint behavior preference value with appropriate default
+                String constraintBehavior = (String) GeneralSharedPreferences.getInstance()
+                        .get(GeneralKeys.KEY_CONSTRAINT_BEHAVIOR);
 
-                    break;
+                // an answer constraint was violated, so we need to display the proper toast(s)
+                // if constraint behavior is on_swipe, this will happen if we do a 'swipe' to the
+                // next question
+                if (constraintBehavior.equals(GeneralKeys.CONSTRAINT_BEHAVIOR_ON_SWIPE)) {
+                    next();
+                } else {
+                    // otherwise, we can get the proper toast(s) by saving with constraint check
+                    saveAnswersForCurrentScreen(EVALUATE_CONSTRAINTS);
                 }
-            }
-        });
 
-        return true;
+                break;
+            }
+        }
+        formSaveViewModel.consumeSavedResult();
```

Comment:
I think we should call this `resumeFormEntry` seeing as that's the "view action" that's occuring. I'm also thinking this should only really happen for results that end the same process (like `SAVED` and the errors) but not for `SAVING` or `CHANGE_REASON_REQUIRED`.

Reply:
Right, fixed.